### PR TITLE
Adds os detection

### PR DESCRIPTION
**Pull request by mlhamel**
## Why

Part of https://github.com/devbuddy/devbuddy/issues/207

We need to have a way to identify the os were devbuddy is running.

## How

This PR is addressing the os detection part of the issue.

### CODE REVIEW

File: pkg/helpers/os/os_linux.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"fmt"
+	"os"
+	"runtime"
+)
+
+// NewOS returns an OS identifier.
+func NewOS() (*OS, error) {
+	variant := "unknown"
+
+	if _, err := os.Stat("/etc/debian_version"); !os.IsNotExist(err) {
```
Comment (merwok) [inline pkg/helpers/os/os_linux.go]: Use `/etc/os-release`, it’s a modern, universal, parseable file format.

Comment (mlhamel) [reply pkg/helpers/os/os_linux.go]: Ideally yes ! but I would prefer not opening a file every time a task is evaluation which could be trigger at every shell hook evaluation.

If we ever support more than debian that would make sense, for now, we just want to know if apt is supported.

File: pkg/helpers/os/os_darwin.go [hunk 2]
```diff
@@ -8,10 +8,10 @@ import (
 )
 
 // NewOS returns an OS identifier.
-func NewOS() (*OS, error) {
+func NewOS() (o *OS, err error) {
 	variant := "unknown"
 
-	if variant, err := syscall.Sysctl("kern.osrelease"); err != nil {
+	if variant, err = syscall.Sysctl("kern.osrelease"); err != nil {
```
Comment (merwok) [inline pkg/helpers/os/os_darwin.go]: Tellement hâte d’avoir un opérateur Python qui change de sens en fonction de deux pixels en plus ou en moins

**Comment by pior** [comment]
On Ubuntu

```
NAME="Ubuntu"
VERSION="17.10 (Artful Aardvark)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 17.10"
VERSION_ID="17.10"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=artful
UBUNTU_CODENAME=artful
```

File: pkg/helpers/os/os_darwin.go (hunk 2 above)
Comment (pior) [reply pkg/helpers/os/os_darwin.go]: a

**Comment by pior** [comment]
Nice source of information: https://unix.stackexchange.com/questions/92199/how-can-i-reliably-get-the-operating-systems-name

File: pkg/helpers/os/os_darwin.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"syscall"
+)
+
+// NewOS returns an OS identifier.
+func NewOS() (o *OS, err error) {
```
Comment (pior) [inline pkg/helpers/os/os_darwin.go]: Just `New` ? 

```go 
os.New()
```

File: pkg/helpers/os/os.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package os
+
+// OS represent the os and it's corresponding release.
+type OS struct {
+	platform string
+	release  string
+}
+
+func NewOSWithRelease(platform string, release string) *OS {
```
Comment (pior) [inline pkg/helpers/os/os.go]: Why do we need that?

File: pkg/helpers/os/os_linux.go (hunk 1 above)
Comment (mlhamel) [reply pkg/helpers/os/os_linux.go]: This work as is but i'm gonna experiment in a follow up PR with https://github.com/acobaugh/osrelease. stay tuned !

**Pull request merged**